prov/gni/src/gnix_msg.c

```diff
@@ -164,10 +180,11 @@ static inline int __gnix_msg_recv_completion(struct gnix_fid_ep *ep,
 	uint64_t flags = FI_RECV | FI_MSG;
 
 	flags |= req->msg.send_flags & (FI_TAGGED | FI_REMOTE_CQ_DATA);
-	flags |= req->msg.recv_flags & (FI_PEEK | FI_CLAIM | FI_DISCARD);
+	flags |= req->msg.recv_flags & (FI_PEEK | FI_CLAIM | FI_DISCARD |
+					FI_MULTI_RECV);
 
 	return __recv_completion(ep, req, req->user_context, flags,
-				 req->msg.recv_len, (void *)req->msg.recv_addr,
+				 req->msg.send_len, (void *)req->msg.recv_addr,
 				 req->msg.imm, req->msg.tag,
```

Comment:
Previously, we would do an extra step of storing the number of moved bytes into the recv_len field right before generating these event.  Now, the request's send_len field should always hold the count of moved bytes when the request is completed.  So the send_len field is now the best place to get this information.